gms/gossiper.cc

```diff
@@ -1841,11 +1841,13 @@ void gossiper::examine_gossiper(utils::chunked_vector<gossip_digest>& g_digest_l
 }
 
 future<> gossiper::start_gossiping(int generation_nbr, std::map<application_state, versioned_value> preload_local_states, gms::advertise_myself advertise) {
-    return container().invoke_on_all([advertise] (gossiper& g) {
-        if (!advertise) {
-            g._advertise_myself = false;
-        }
-    }).then([this, generation_nbr, preload_local_states] () mutable {
+    return seastar::async([this, g = this->shared_from_this(), generation_nbr, preload_local_states, advertise] () mutable {
```

Comment:
why do you need seastar::async ? it doesn't seem required as you seem to be already in an async context.

Reply:
To use get(), I need to be inside async http://docs.seastar.io/master/tutorial.html#saving-objects-on-the-stack